src/session.ts

```diff
@@ -462,4 +451,45 @@ function getScreenHint(signUpPaths: string[] | undefined, pathname: string) {
   return screenHintPaths.length > 0 ? 'sign-up' : 'sign-in';
 }
 
+/**
+ * Saves a WorkOS session to a cookie for use with AuthKit.
+ *
+ * This function is intended for advanced use cases where you need to manually manage sessions,
+ * such as custom authentication flows (email verification, etc.) that don't use
+ * the standard AuthKit authentication flow.
+ *
+ * @param session The WorkOS session containing access token, refresh token, and user information.
+ * @param request Either a NextRequest object or a URL string, used to determine cookie settings.
+ *
+ * @example
+ * // With a NextRequest object
+ * import { saveSession } from '@workos-inc/authkit-nextjs';
+ *
+ * async function handleEmailVerification(req: NextRequest) {
+ *   const { code } = await req.json();
+ *   const authResponse = await workos.userManagement.authenticateWithEmailVerification({
+ *     clientId: process.env.WORKOS_CLIENT_ID,
+ *     code,
+ *   });
+ *
+ *   await saveSession({
+ *     accessToken: authResponse.accessToken,
+ *     refreshToken: authResponse.refreshToken,
+ *     user: authResponse.user,
+ *     impersonator: authResponse.impersonator
+ *   }, req);
+ * }
+ *
+ * @example
+ * // With a URL string
+ * await saveSession(session, 'https://example.com/callback');
+ */
+export async function saveSession(session: Session, request: NextRequest | string): Promise<void> {
```

Comment:
this looks great, i wonder if saveSession should take an `AuthenticationResponse` instead of a `Session` (they're probably the same shape, but i think anytime you'd call this you just want to feed it the response of `authenticateWithFoo` ?

Reply:
Yeah, looks like they're roughly the same shape. Thinking of changing the signature to this just for self-documenting:

```ts
export async function saveSession(
  sessionOrResponse: Session | AuthenticationResponse,
  request: NextRequest | string,
): Promise<void>;
```